### DISCUSSION TITLE
Upgrade to 2023.7 core breaks Nilan connection

### DISCUSSION THREAD

**Discussion by ollila** [veista/nilan#72] [Q&A] [answered]
I have Nilan EC9 with CT602 mobus v22 connected to HA (Core 2023.6.3) through Waveshare RS485-to-Eth working as intended. When I upgrade to 2023.7 series core something breaks and connection between HA and Nilan is lost. The issue is replicable and connection returns when reverting to 2023.6 core from backups.

**Comment by veista** [comment]
Hi, Which version of the integration are you running?

**Comment by ollila** [comment]
I'm currently running 1.1.17

**Comment by veista** [comment]
Have you tried the beta 1.2.1? I'm running that with newest HA and it is running fine on my setup.

**Comment by ollila** [comment]
I tried 1.2.1 and managed to get it running under 2023.6 (although I messed sensor names). When updating to 2023.7.3 it loses connection to Nilan. From HA logs I see the following error messages:

Could not read get_bus_version
6:21:33 PM – (ERROR) Nilan (custom integration)
Could not read get_machine_type
6:21:31 PM – (ERROR) Nilan (custom integration)
Pymodbus: Nilan: Modbus Error: [Input/Output] No Response received from the remote slave/Unable to decode response
6:21:31 PM – (ERROR) Modbus

Since pymodbus was updated in 2023.7 and produces error here I would guess these are linked.

**Comment by veista** [comment]
Have you tried restarting the bridge device after updating?

**Comment by ollila** [comment]
I will try it next but it will most likely slip to weekend before I have enough time in case I need to restore installation from back up.

**Comment by ollila** [comment]
I tried the bridge restart without success this time with 2023.8 core. After restart bridge has only the link light on (green) but not the blue indicating bridge has not connected to host. I enabled debug logging before update and the only usefult messages are what I already saw from HA logs:

2023-08-04 20:25:48.153 ERROR (SyncWorker_4) [homeassistant.components.modbus.modbus] Pymodbus: Nilan: Modbus Error: [Input/Output] No Response received from the remote slave/Unable to decode response
2023-08-04 20:25:49.511 ERROR (MainThread) [custom_components.nilan.device] Could not read get_machine_type
2023-08-04 20:25:49.603 ERROR (MainThread) [custom_components.nilan.device] Could not read get_bus_version

Here is the bridge configuration in case it makes any difference:
![image](https://github.com/veista/nilan/assets/113829131/555c91e5-12b7-481e-8e4b-ea43f84fa766)

**Comment by veista** [comment, marked as answer]
I don't have this bridge my self but your work mode should be server, not client.

**Comment by ollila** [comment]
Changing  destination IP to the HA server IP and changing work mode to server fixed the issue, thank you! I wonder how I ever managed to get it running with wrong configuration.

**Reply by veista** [reply]
I was wondering that too. Good that the issue got resolved.

**Reply by MKBU2** [reply]
This also fixed my identical issue with the same RS485 bridge.
